api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java

```diff
@@ -269,11 +270,38 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
 
     @Override
     public <T> Boolean in(BoundReference<T> ref, Set<T> literalSet) {
+      Integer id = ref.fieldId();
+
+      if (containsNullsOnly(id)) {
+        return ROWS_CANNOT_MATCH;
+      }
+
+      final Comparator<T> comparator = ((BoundSetPredicate<T>) expr).comparator();
+      Set<T> literals = literalSet;
+
+      if (lowerBounds != null && lowerBounds.containsKey(id)) {
+        T lower = Conversions.fromByteBuffer(ref.type(), lowerBounds.get(id));
+        literals = literals.stream().filter(v -> comparator.compare(lower, v) <= 0).collect(Collectors.toSet());
```

Comment:
There's no need to use a set to collect the values, and the set would not correctly handle mixed UTF8 string types.

I think it would be better to use a list for the result. That would mean that `literals` can't be reused, but I think it is fine to use a separate `List<T>` variable for values that are greater than the lower bound.

Reply:
Done.